infra/blueprint-test/README.md

```diff
@@ -0,0 +1,409 @@
+- `test/fixtures`  - this directory contains “fixture” configuration. In most cases, this should be configured to wrap examples that need additional inputs and unify the interface for testing. Usage of fixtures is discouraged unless necessary.
+- `test/integration`  - this directory is intended to hold integration tests that are responsible for running and asserting test values for a given fixture.
+- `test/setup`  - this directory holds the GCP project and initial resource setup modules that are a prerequisite for the blueprint tests to run.
+
+# 3. Test Development
+This section aims at explaining the process of developing a custom integration test which consists of the following steps:
+
+## 3.1 Terraform based blueprints
+
+### 3.1.1 Prepare a test example 
+The first step in the process is to create an example that leverages a TF module from the blueprint as illustrated in section 2.1. The example creation process consists of two critical steps:
+
+#### 3.1.1.1 Create the example configuration
+In this step you will create an example directory under the `examples` directory that uses a module or submodule from the `modules` as the source as follows:
+1. cd into `examples` directory
+2. Choose a name, such as `mysql-public` for the example and create a new directory with that name and cd into it
+3. Create a new file named `main.tf` with the following content
+
+```hcl
+// name for your example module
+module "mysql-db" { 
+ // set the source for the module being tested as part of the 
+ // example
+ source               = "../../modules/mysql"
+ ...
+}
+```
+
+4. Create a new file named `variables.tf` for organizing and defining variables that need to be passed into the example module as follows:
+
+```hcl
+...
+variable "project_id" {
+ description = "The ID of the project in which resources will be provisioned."
+ type        = string
+}
+ 
+variable "db_name" {
+ description = "The name of the SQL Database instance"
+ default     = "example-mysql-public"
+}
+...
+```
+
+These variables are now available for access within the `main.tf` file and can be set in the example module. The following example shows how to reference the `project_id` variable in `main.tf`.
+
+```hcl
+// name for your example module
+module "mysql-db" { 
+ ...
+ // variables required by the source module
+ random_instance_name = true
+ database_version     = "MYSQL_5_6"
+ 
+ // variable being set from the example module variables configuration. 
+ project_id           = var.project_id
+ ...
+}
+```
+
+```
+Note:
+Variables defined in the example module’s variables (normally variables.tf) configuration can be set 1) from 
+upstream modules calling the example module (e.g. fixtures) or 2) using environment variables by prefixing 
+the environment variable with “TF_VAR”. E.g. to set the project_id variable (above), setting the value in a 
+“TF_VAR_project_id” environment variable would automatically populate its value upon execution.
```

Comment:
We should mention `test/setup` and how the framework exports outputs as env vars from here

Reply:
Added. Let me know that makes sense to you.